src/aiortc/rtcrtpsender.py

```diff
@@ -458,6 +374,16 @@ async def _run_rtcp(self) -> None:
                 # range [0.5, 1.5] times the calculated interval.
                 await asyncio.sleep(0.5 + random.random())
 
+                if self.__prev_ntp_timestamp == self.__ntp_timestamp:
+                    """
+                    Safety mechanism: After the video has been fully sent,
+                    self.__ntp_timestamp is no longer updated in run_rtp which
+                    casuses a lot of sender's RTCP packets to have the same lsr.
+                    Having the same lsr produces error in identifying which received
+                    RTCP packet matches which sent RTCP packet (producing negative rtt).
+                    """
```

Comment:
Are we sure this only executes at the end? Otherwise, we might end up making a lot of redundant NTP calls and I don't know how expensive they are

Reply:
yes, I'm sure it happens after the sender's video has finished (the experiment is not necessarily finished when the throughput is not 30 fps). Happy to change this solution to something else if there is an easy alternative.